### ISSUE TITLE
Simple filter

### ISSUE THREAD

**Issue by ltguillaume** [GM-Script-Writer-62850/HTML5-Music-Player#2]
Another thing that would greatly improve the experience is to have a filter field. Pseudo-code-ish:

If filter field not empty:
1. Unfold tree (height:auto;)
2. If not entry.contains(filter) then height:0;

**Comment by GM-Script-Writer-62850** [comment]
What would this filter match track and/or folder?
i could add a search feature or perhaps a find feature to open all folders that contain a match

I did think about this but i did not want to clutter up the UI (even the clear played list button is questionable IMO)
I could add a keyboard control to open a prompt, but mobile users would have a hard time accessing that, any suggestions?

BTW I do not use JQuery in this project, everything is 100% native JavaScript

My goal was to make a simple music player that just plays music, no fancy special features, just simple and to the point

**Comment by ltguillaume** [comment]
Yes, I meant that it should match both folder and filename. Basically it
could just use the same height style property to collapse/unfold like
you're using now (like I suggested in the pseudo-code). It would take a
single input field and a listener to check for changes in that input field.

A search/filter field can be made very inconspicuous, so I don't think it
would clutter the UI. Otherwise, you could create a "hamburger menu" that
harbors the field, and the clear played button.

I think jQuery is highly overrated, and overused for smaller projects, so
I'm happy you're not using it for this little gem :-)

**Comment by GM-Script-Writer-62850** [comment]
If you wanted to make something yourself, here is a simple search function
```
function findTracks(src){// returns array of indexes of matches
	var res=[],i;
	src=new RegExp(src,'i');// i means ignore case
	for(i in music){
		if(music[i].getAttribute('file').match(src))
			res.push(i);
	}
  return res;
}
var musicIndexes=findTracks('string to find');
```
you could do something like make a list of matches or open all matches in the playlist so you can scroll to them

this will match anything the file path, including the name used for the library folder

**Comment by ltguillaume** [comment]
Thanks! I've just finished writing my take on such a folder player. Although written from scratch, a lot of it is based on your excellent project, such as some shortcuts in serialization and quite some structural stuff. Credits in readme.md. https://github.com/ltGuillaume/MusicFolderPlayer

It's probably not up to par compatibility-wise (MIME types for example), but I have yet to encounter any issues with playback.

**Comment by GM-Script-Writer-62850** [comment]
looks good at least in your screnshot, be careful using the Segoe UI fonts, not cross platform
looks like you made your own ui for the player, i have made that but not committed it, still have not fixed a font glitch in ms edge

as for the mime types, there are some cases where mp3 for example is not supported due to licensing issues, eg firefox on devices that do not have mp3 supported included in the os, the license fee is absurd

**Comment by ltguillaume** [comment]
- It'll fall back on a sans-serif font just fine. Pretty much all CSS dimensions are in `em`, so a different font won't break the UI.
- I wrote everything from scratch, but as said, I really liked the elegance of some techniques you used. I basically mirrored the way you populate the tree view (PHP and JavaScript bit). I haven't looked at your implementation of `hst`, since I had my own preferences for the entire playlist / playback dynamics. For example, I'm assuming that in between sessions files could be added to/removed from the library, so I'm saving the paths in the Local Storage, not just id's.
- I think your current commit breaks when Local Storage is disabled. Have a look at the function ls() I wrote.
- Good point about MP3, although since 2012 in the EU, and 2017 in the US, it became patent-free.
